Export Analysis Reports/Sdl.Community.ExportAnalysisReports/ReportExporterControl.cs

```diff
@@ -577,45 +577,49 @@ private void GenerateReport()
 		{
 			try
 			{
-				if (!IsNullOrEmpty(reportOutputPath.Text) && Directory.Exists(reportOutputPath.Text))
+				if (!IsNullOrEmpty(reportOutputPath.Text))
 				{
+					Help.CreateDirectory(reportOutputPath.Text);
 					var projectsToBeExported = _projectsDataSource.Where(p => p.ShouldBeExported).ToList();
-					foreach (var project in projectsToBeExported)
+					var areCheckedLanguages = projectsToBeExported.Any(p => p.LanguagesForPoject.Any(l => l.Value));
+					if (!areCheckedLanguages && projectsToBeExported.Count >= 1)
 					{
-						// check which languages to export
-						var checkedLanguages = project.LanguagesForPoject.Where(l => l.Value).ToList();						
-						if(checkedLanguages.Count == 0 && projectsToBeExported.Count == 1)
-						{
-							_messageBoxService.ShowOwnerInformationMessage(this, "Please select at least one language to export the report!", "Export result");
-							return;
-						}
-						foreach (var languageReport in checkedLanguages)
+						_messageBoxService.ShowOwnerInformationMessage(this, "Please select at least one language to export the report!", "Export result");
+					}
+					else
+					{
+						foreach (var project in projectsToBeExported)
 						{
+							// check which languages to export
+							var checkedLanguages = project.LanguagesForPoject.Where(l => l.Value).ToList();
```

Comment:
we can move this where outside the foreach, where the "areCheckedLanguages" variable created. and in the validation we only take the count. 
In this way we only iterate the list once to see the count and also to get the languages selected.

Reply:
The variable cannot be removed because depends on each project which is iterated through foreach